### PR TITLE
Add steps to test main to correctly init gmock

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Currently we do not init GMock correctly, as per
https://chromium.googlesource.com/external/github.com/google/googletest/+/refs/tags/release-1.8.0/googlemock/docs/ForDummies.md

This adds the missing steps, and removes a file which does not partake
in any sort of main generation.

By adding this ctest will correctly mark GMock expectation failures as
failures instead of passing if TS_ASSERT(VERIFY_AND_CLEAR(x)) pattern is
not used or missed

**To test:**
- Check tests pass
- Change an EXPECT_CALL in a random test to an incorrect value (that does not have an explicit verify) you should see CTest correctly fail it now

Part of #31069 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h [hunk 1]
```diff
@@ -37,74 +67,43 @@ class AlgorithmProgressDialogPresenterTest : public CxxTest::TestSuite {
...
-    // but the algorithm will trigger them too
-    // adding the expected calls prevents a bunch of GMock warnings
-    EXPECT_CALL(*mainProgressBar.get(), algorithmStarted()).Times(1);
-    for (const auto prog : {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0}) {
-      EXPECT_CALL(*mainProgressBar.get(), updateProgress(DoubleEq(prog), emptyQString, 0.0, 0));
-    }
-    EXPECT_CALL(*mainProgressBar.get(), algorithmEnded()).Times(1);
-    // End of assertions for the main progress bar
-    //----------
+  void xtestAlgorithmIsTrackedCorrectly() {
```
Comment (martyngigg) [inline qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h:75]: Did we mean to leave an `x` at the start of this test name? I think this disables to test?

File: qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h [hunk 2]
```diff
@@ -37,74 +67,43 @@ class AlgorithmProgressDialogPresenterTest : public CxxTest::TestSuite {
...
+    TS_ASSERT_EQUALS(size_t{0}, mockedTypes.m_presenter->getNumberTrackedAlgorithms());
   }
   /** This tests running algorithms from inside an algorithm
    * and that they are all properly tracked for their lifetime
    * in the dialog
    */
-  void testAlgorithmThatRunsOtherAlgorithmsIsTrackedCorrectly() {
-    auto mainProgressBar = mockDialogView->mainProgressBar;
-
+  void xtestAlgorithmThatRunsOtherAlgorithmsIsTrackedCorrectly() {
```
Comment (martyngigg) [inline qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h:102]: `x` here...

**Review by martyngigg — CHANGES_REQUESTED** [review]
Good catch on the missing `VerifyAndClear`. It's mildly scary they have been missing for so long.

I noticed a couple of tests had been disabled but had been left in probably by accident?

**Comment by martyngigg** [comment]
Look like we have a merge conflict here too...

File: qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h [hunk 3]
```diff
@@ -37,74 +67,43 @@ class AlgorithmProgressDialogPresenterTest : public CxxTest::TestSuite {
     delete suite;
   }
 
-  void setUp() override {
-    mockDialogView.reset();
-    // The mock view also creates the presenter, because
-    // so that is passes the correct type into the constructor
-    mockDialogView = std::make_unique<NiceMock<MockAlgorithmProgressDialogWidget>>();
-  }
+  void test_true() { TS_ASSERT(true); }
```
Comment (martyngigg) [inline qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h:70]: Is this also a relic of debugging?

Comment (DavidFair) [reply qt/widgets/common/test/AlgorithmProgress/AlgorithmProgressDialogPresenterTest.h:70]: Yes I thought I'd removed it but a rebase brought it back during conflict resolution :man_facepalming: 
Thanks for noticing

**Review by martyngigg — APPROVED** [review]
Looks good!

**Pull request merged**